smriprep/workflows/anatomical.py

```diff
@@ -1049,9 +1086,187 @@ def init_anat_fit_wf(
         ])
         # fmt:on
     elif not t2w:
-        LOGGER.info("No T2w images provided - skipping Stage 7")
+        LOGGER.info("ANAT No T2w images provided - skipping Stage 7")
+    else:
+        LOGGER.info("ANAT Found preprocessed T2w - skipping Stage 7")
+
+    # Stages 8-10: Surface conversion and registration
+    # sphere_reg is needed to generate sphere_reg_fsLR
+    # sphere and sulc are needed to generate sphere_reg_msm
+    # white, pial, midthickness and thickness are needed to resample in the cortical ribbon
+    # TODO: Consider paring down or splitting into a subworkflow that can be called on-demand
+    # A subworkflow would still need to check for precomputed outputs
+    needed_anat_surfs = ["white", "pial", "midthickness"]
+    needed_metrics = ["thickness", "sulc"]
+    needed_spheres = ["sphere_reg", "sphere"]
```

Comment:
seems like a good time for sets

```suggestion
    needed_anat_surfs = {"white", "pial", "midthickness"}
    needed_metrics = {"thickness", "sulc"}
    needed_spheres = {"sphere_reg", "sphere"}
```

Reply:
Sets can vary in order with each interpreter, so I'd worry about this causing some node to consider its inputs changed. The n is so small here that I think efficiency is not a significant concern.